api/src/main/scala/Geoprocessing.scala

```diff
@@ -47,6 +47,63 @@ trait Geoprocessing extends Utils {
     ResultDouble(average)
   }
 
+  def getRasterLinesJoin(input: InputData): ResultInt = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
+    val lines = input.vector match {
+      case Some(vector) =>
+        input.vectorCRS match {
+          case Some(crs) =>
+            createMultiLineFromInput(vector, crs, input.rasterCRS)
+          case None =>
+            throw new Exception("Request data missing required 'vectorCRS'.")
+        }
+      case None =>
+        throw new Exception("Request data missing required 'vector'.")
+    }
+    ResultInt(rasterLinesJoin(rasterLayers, lines))
+  }
+
+  /**
+    * Given a collction of rasterLayers & a collection of lines, return the
+    * values intersected by the rasterized lines.
+    *
+    * @param   rasterLayers  A sequence of TileLayerCollections
+    * @param   lines         A sequence of MultiLines
+    * @return                A map of pixel counts
+    */
+  private def rasterLinesJoin(
+    rasterLayers: Seq[TileLayerCollection[SpatialKey]],
+    lines: Seq[MultiLine]
+  ): Map[String, Int] = {
+    val metadata = rasterLayers.head.metadata
+    var pixelGroups: TrieMap[(List[Int], List[Int]), Int] = TrieMap.empty
+
+    joinCollectionLayers(rasterLayers).par
+      .foreach({ case (key, tiles) =>
+        val extent = metadata.mapTransform(key)
+        val re = RasterExtent(extent, metadata.layout.tileCols,
+            metadata.layout.tileRows)
+
+        lines.par.foreach({ multiLine =>
+          Rasterizer.foreachCellByMultiLineString(multiLine, re) { case (col, row) =>
+            val pixelGroup: (List[Int], List[Int]) =
+              (tiles.map(_.get(col, row)).toList, List(col, row))
```

Comment:
Just occurred to me that storing the col/row as part of the TrieMap key like this could lead to undercounting if it's the case that multiple tiles would have the same `col, row` pair in them. Potentially this can be avoided by storing the tile key in the key tuple, too.

Reply:
This turned out to be the problem.